### DISCUSSION TITLE
📎 What would you like to see next in our Documentation ?

### DISCUSSION THREAD

**Discussion by erogol** [coqui-ai/TTS#799] [Ideas]
Here we need your feedback on 🐸 TTS Documentation.

Let us know what you think about our documentation https://tts.readthedocs.io/en/latest/ and your recommendations or expectations to improve it.

All kinds of feedback are welcome!!

**Comment by ZDisket** [comment]
More emphasis on finetuning. Transfer learning greatly reduces data, time, and compute costs, lowering the barrier for entry.

**Reply by yuananf** [reply]
If possible, a training tutorial like this is very good
https://awesomeface349.gitbook.io/uberduck-written-tutorial/training-time/simplified-training-page

And document what languages this repo supports, how to train them each.

**Reply by erogol** [reply]
Have you seen these? 

https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html
https://tts.readthedocs.io/en/latest/training_a_model.html

If so what do you think is missing there?

**Reply by yuananf** [reply]
> Have you seen these?
> 
> https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html
> https://tts.readthedocs.io/en/latest/training_a_model.html
> 
> If so what do you think is missing there?

Yes I saw this, but it looks like more general, if more details are provided, that will be great. 
For example, 
1. What should I do if I want to finetune a model using a pretrained model. Like @ZDisket said.
2. What parameter should I pass if I want to continue a training process, I found the parameter in the code and issue but not in the doc, and there is a bug with this parameter. I removed the ```f``` so I can continue training https://github.com/coqui-ai/TTS/blob/main/TTS/trainer.py#L881.
3. I saw there are tips about train in another language, it will be great with examples provided.

**Reply by KakoozaJerry** [reply]
I also think if they could provide some parameters we can look out for to know if we are achieving a good model for example for tacotron, maybe talk of the tensor board logs and which ones will tell you that you are moving in the right direction, whether the loss curves since there are so many loss curves at different stages, also the attention/alignment model graph of encoder vs decoder. The one that leads to a diagonal for the tacotron 2. I think some insights into specific models best graphs or number of steps as you are most likely to start getting some good output plus a rough idea on how long it would take training specific dataset sizes like LJ Speech dataset, Rough idea on real world scenarios.

**Reply by erogol** [reply]
Some of these are addressed here https://tts.readthedocs.io/en/latest/faq.html

but needs more info to precisely get there

Maybe adding tensorboard logs to the model released might also help but in general, logs are larger than the model :)

**Comment by erogol** [comment]
I wrote some docs for finetuning. Any feedback is welcome. 

https://tts.readthedocs.io/en/latest/finetuning.html

**Reply by yuananf** [reply]
> I wrote some docs for finetuning. Any feedback is welcome.
> 
> https://tts.readthedocs.io/en/latest/finetuning.html

This is great!

**Comment by thoraxe** [comment]
I'd like to see some examples of minimum viable utterance/sentence/recording sets for producing a model (English). There are the general guidelines, but the guidelines are not specific enough for a non-practitioner to understand what things like "Good phoneme coverage. Make sure that your dataset covers a good portion of the phonemes, di-phonemes, and in some languages tri-phonemes." actually mean.

In the absence of an example, I went searching around for something and didn't come up with much. I posted in the Mozilla TTS discourse:
https://discourse.mozilla.org/t/minimum-viable-english-phoneme-coverage-set-ljspeech-style/86854

They pointed me at:
https://github.com/rhasspy/tts-prompts

This is better than nothing but is still 1,1100 sentences (English). And I'm not entirely clear how truly balanced it is:

![phoneme](https://user-images.githubusercontent.com/94413/136450436-7ccb7af4-4541-42f8-a997-d9309a1b1286.png)

I haven't found much research on this topic, so I've started pinging researchers. A more solid "start here with these 100 sentences" example for custom voice/model training would be an epic addition to the documentation, and I'd be willing to help as best I can being a non-data-scientist.
